src/pages/AdminEnrollmentList.js

```diff
@@ -142,19 +151,49 @@ class AdminEnrollmentList extends React.Component {
       style: enrollmentListStyle.cell,
       width: 130,
       filterable: true,
-      Filter: ({ filter, onChange }) => (
-        <select
-          onChange={event => onChange(event.target.value)}
-          value={filter ? filter.value : ''}
-        >
-          <option value="">Tous</option>
-          {toPairs(TARGET_API_LABELS).map(([key, label]) => (
-            <option key={key} value={key}>
-              {label}
-            </option>
-          ))}
-        </select>
-      ),
+      Filter: ({ filter, onChange }) => {
+        // Note that users own enrollments might not be available through this filter
+        const availableTargetApi = this.props.user.roles
+          .filter(role => role.endsWith(':reporter'))
+          .map(role => {
+            const targetApiKey = role.split(':')[0];
+
+            return {
+              key: targetApiKey,
+              label: TARGET_API_LABELS[targetApiKey],
+            };
+          });
+
+        const value = filter
+          ? isArray(filter.value)
+            ? filter.value
+            : [filter.value]
```

Comment:
double ternary is very hard to read. I wonder what are the input and output signature now